### PR TITLE
[#160426807] automatic SPID session token retrieval at setup

### PR DESCRIPTION
**Pull request by balanza**
This implements automatic session token resolution. It's done by scraping `spid-test-env` injecting given `username` and `password` for a test user. 
The following environment variables are introduced:
* SPID_LOGIN_HOST
* SPID_USERNAME
* SPID_PASSWORD

In this first implementation it performs a level-1 Login with no options.

##### known issues
1. Data is passed from the setup workflow to the test cases by **modifying `process.env`**. This is because of an unresolved issue on `jest`: https://github.com/facebook/jest/issues/2549.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,8 +11,24 @@ We need to test the API ecosystem behind [IO-app](https://github.com/pagopa/io-a
 ## Usage
 ```
 yarn install
+yarn generate
 yarn start 
 ```
 
+You'll be propted with a required input in which you can specify reference for the environment to test. Every input item is overridable by passing its specific env variable:
```
Comment (gunzip) [inline README.md]: may you please use this table format to document env variables:
ie. https://github.com/pagopa/io-backend#environment-variables

File: setup/index.ts [hunk 2]
```diff
@@ -1,11 +1,125 @@
...
+        })
+      ),
+    value => tryCatch(() => Promise.resolve(value), toError)
+  );
+
+  /*   username: "alice.rossi",
+  password: "io-app-test",
+  host: "https://app-backend.dev.io.italia.it" */
+
+  /*   const SPID_LOGIN_HOST = "https://app-backend.dev.io.italia.it";
```
Comment (gunzip) [inline setup/index.ts]: use `//` to format inline comments or block comments `/** */` to describe methods

Comment (gunzip) [reply setup/index.ts]: it this is an example write some comments other than the code (otherwise: do not commit commented out code)

**Comment by gunzip** [comment]
can you please use an english title for this PR that describes what it does ?

File: setup/index.ts (hunk 2 above)
Comment (balanza) [reply setup/index.ts]: Yes, it wasn't intended

**Comment by gunzip** [comment]
just note that since we'll need to run tests into the production environment, this feature adds little value since we won't have a spid test environment there

**Comment by balanza** [comment]
good to know 😅 

> just note that since we'll need to run tests into the production environment, this feature adds little value since we won't have a spid test environment there (as said, we will use a pre-defined token)

**Pull request merged**